ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
-        "layer are: activation, dropout, norm, norm_params, output_size, use_bias, bias_initializer and "
-        "weights_initializer. If any of those values is missing from the dictionary, the default one "
-        "specified as a parameter of the encoder will be used instead. ",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["fc_layers"],
+
+@register_encoder_config("efficientnet_torch", IMAGE)
+@dataclass
+class TVEfficientNetEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["efficientnet_torch"],
+        default="efficientnet_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        [
+            "b0",
+            "b1",
+            "b2",
+            "b3",
+            "b4",
+            "b5",
+            "b6",
+            "b7",
+            "v2_",
+            "v2_m",
+            "v2_l",
+        ],
+        default="b0",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVEfficientNetEncoder"]["model_variant"],
+    )
 
-@register_encoder_config("mlp_mixer", IMAGE)
-@dataclass(repr=False)
-class MLPMixerEncoderConfig(BaseEncoderConfig):
 
+@register_encoder_config("googlenet_torch", IMAGE)
+@dataclass
+class TVGoogLeNetEncoderConfig(TVBaseEncoderConfig):
     type: str = schema_utils.StringOptions(
-        ["mlp_mixer"],
-        default="mlp_mixer",
+        ["googlenet_torch"],
+        default="googlenet_torch",
         allow_none=False,
         description="Type of encoder.",
     )
 
-    dropout: float = schema_utils.FloatRange(
-        default=0.0,
-        min=0,
-        max=1,
-        description="Dropout rate.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["dropout"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        ["base"],
+        default="base",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVGoogLeNetEncoder"]["model_variant"],
     )
 
-    height: int = schema_utils.NonNegativeInteger(
-        default=None,
-        description="Height of the input image.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["height"],
+
+@register_encoder_config("inceptionv3_torch", IMAGE)
+@dataclass
+class TVInceptionV3EncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["inceptionv3_torch"],
+        default="inceptionv3_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    width: int = schema_utils.NonNegativeInteger(
-        default=None,
-        description="Width of the input image.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["width"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
```

Comment:
Want to double-check if this field is really `Optional`? If so, make sure `allow_none` is `True`.

Reply:
removed optional